OneCommunityGlobal/HighestGoodNetworkApp
# Nathan permission categories

# Description
This PR turns category permissions into multi-checkboxes instead of their own permissions to prevent overlap. To do this, it creates a component to display permissions wherever needed

## Related PRS (if any):
[Back-end PR 585 fixes the `Cannot read property 'permissions' of null` BE error which will make testing this easier](https://github.com/OneCommunityGlobal/HGNRest/pull/585)

## Main changes explained:
- Created `PermissionsConst` to restructure permissions to account for categories and consolidate `mainPermisisons`, `modalInfo`, and `permissionLabel`.
- Created `PermissionList` and `PermissionListItem` components to be reused wherever needed
- Added multi-checkbox functionality to categories in `PermissionListItem`
- Moved info point modal logic into `PermissionListItem` to reduce redundancy
- Implemented `PermissionList` for updating role permissions, role presets, new roles, and user permissions

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. login as owner user
4. go to dashboard→ Other Links→ Permissions Management
5. go to each of: "Manage User Permissions", "Add New Role", any actual role, and "Load Presets" in any role
6. verify info points work in every location
7. verify adding and removing permissions work in every location (except presets which should not be editable)
8. verify changes save and persist after reload

## Screenshots or videos of changes:
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/23086054/04e1e1d3-27ad-4e9a-901b-8d20a2468f56
